From 2f465e8b0b72e7a8770c1ff25d59e83087e9dba3 Mon Sep 17 00:00:00 2001
From: Chris Blake <chrisrblake93@gmail.com>
Date: Thu, 4 Oct 2018 20:49:01 -0500
Subject: [PATCH 02/10] watchdog: Fix SPL builds when watchdog is enabled

Without this fix, u-boot fails to comple as it tries to load watchdog.h
during an SPL build. This breaks support for things such as the sunxi
arch.

lib/built-in.o: In function `udelay': ./u-boot/lib/time.c:167: undefined
reference to `hw_watchdog_reset'
drivers/built-in.o: In function `NS16550_putc':
./u-boot/drivers/serial/ns16550.c:217: undefined reference to
`hw_watchdog_reset'
drivers/built-in.o: In function `NS16550_getc':
./u-boot/drivers/serial/ns16550.c:228: undefined reference to
`hw_watchdog_reset'
scripts/Makefile.spl:353: recipe for target 'spl/u-boot-spl' failed
make[1]: *** [spl/u-boot-spl] Error 1
Makefile:1508: recipe for target 'spl/u-boot-spl' failed
make: *** [spl/u-boot-spl] Error 2
---
 include/watchdog.h | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/watchdog.h b/include/watchdog.h
index a4a4e8e614..29061256bc 100644
--- a/include/watchdog.h
+++ b/include/watchdog.h
@@ -20,7 +20,7 @@
 int init_func_watchdog_reset(void);
 #endif
 
-#if defined(CONFIG_WATCHDOG) || defined(CONFIG_HW_WATCHDOG)
+#if (defined(CONFIG_WATCHDOG) || defined(CONFIG_HW_WATCHDOG)) && !defined(CONFIG_SPL_BUILD)
 #define INIT_FUNC_WATCHDOG_INIT	init_func_watchdog_init,
 #define INIT_FUNC_WATCHDOG_RESET	init_func_watchdog_reset,
 #else
@@ -35,7 +35,7 @@ int init_func_watchdog_reset(void);
 /*
  * Hardware watchdog
  */
-#ifdef CONFIG_HW_WATCHDOG
+#if defined(CONFIG_HW_WATCHDOG) && !defined(CONFIG_SPL_BUILD)
 	#if defined(__ASSEMBLY__)
 		#define WATCHDOG_RESET bl hw_watchdog_reset
 	#else
@@ -47,7 +47,7 @@ int init_func_watchdog_reset(void);
 	/*
 	 * Maybe a software watchdog?
 	 */
-	#if defined(CONFIG_WATCHDOG)
+	#if defined(CONFIG_WATCHDOG) && !defined(CONFIG_SPL_BUILD)
 		#if defined(__ASSEMBLY__)
 			#define WATCHDOG_RESET bl watchdog_reset
 		#else
-- 
2.17.1

